### PR TITLE
fix: Correct a normalization regex to fix a flaky test, print query test diffs in CI

### LINKED ISSUE #6819: Flaky test query_tests2::cases::dedup_and_predicates_parquet_ingester
Haven't been able to get this to fail locally. Investigating.

### PR DESCRIPTION
**Pull request by carols10cents**
Closes #6819.

## What happened

I observed `query_tests2::cases::dedup_and_predicates_parquet_ingester` fail in CI only, on branches that contained https://github.com/influxdata/influxdb_iox/pull/6807. Example: [crepererum/revert_part_throttling](https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/28801/workflows/d32d878e-0862-4829-a4ba-f810bcfa3678/jobs/259460)

The SQL file based tests don't print out the diff because you might need to update the `.expected` file. But in CI it's annoying (impossible?) to get the files from disk, and you won't be updating the `.expected` files in CI anyway. So I changed that test code to always print out the diff of the files when the test fails in CI-- that's the first commit in this branch.

Then, once I could see the reason the test was failing, the diff [in the test output](https://circleci.com/api/v1.1/project/github/influxdata/influxdb_iox/259564/output/104/0?file=true&allocation-id=63dbea4381083c4b9ef1da39-0-build%2F76065175) showed this:

```diff
-|               |               ParquetExec: limit=None, partitions={1 group: [[1/1/1/1/00000000-0000-0000-0000-000000000000.parquet]]}, output_ordering=[tag@2 ASC, time@3 ASC], projection=[bar, foo, tag, time] |
+|               |               ParquetExec: limit=None, partitions={1 group: [[1/1/1/10/00000000-0000-0000-0000-000000000000.parquet]]}, output_ordering=[tag@2 ASC, time@3 ASC], projection=[bar, foo, tag, time] |
```

Note that the filename in the expected file is `1/1/1/1/` and the filename in the actual file is `1/1/1/10/`.

The test code is supposed to normalize these filepaths, buuuut I messed up that regex such that it would only handle one digit at the end :( And now that we have lots of tests using this framework, sometimes the directory has two digits.

The second commit fixes the regex and should fix the flaky test.

### CODE REVIEW

File: test_helpers_end_to_end/src/snapshot_comparison.rs [hunk 1]
```diff
@@ -191,7 +215,7 @@ async fn run_query(cluster: &MiniCluster, query: &Query) -> Result<Vec<String>>
     if query.normalized_uuids() {
         let regex_uuid = Regex::new("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}")
             .expect("UUID regex");
-        let regex_dirs = Regex::new(r#"[0-9]+/[0-9]+/[0-9]+/[0-9+]"#).expect("directory regex");
+        let regex_dirs = Regex::new(r#"[0-9]+/[0-9]+/[0-9]+/[0-9]+"#).expect("directory regex");
```
Comment (crepererum) [inline test_helpers_end_to_end/src/snapshot_comparison.rs:218]: :facepalm:

**Review by crepererum — APPROVED** [review]

File: test_helpers_end_to_end/src/snapshot_comparison.rs [hunk 2]
```diff
@@ -98,14 +98,38 @@ pub async fn run(
...
-        println!("Possibly helpful commands:");
-        println!("  # See diff");
-        println!("  diff -du {:?} {:?}", expected_path, output_path);
-        println!("  # Update expected");
-        println!("  cp -f {:?} {:?}", output_path, expected_path);
+        if std::env::var("CI")
+            .map(|value| value == "true")
+            .unwrap_or(false)
+        {
+            // In CI, print out the contents because it's inconvenient to access the files and
```
Comment (alamb) [inline test_helpers_end_to_end/src/snapshot_comparison.rs:105]: 👍 I wonder if we should also revert https://github.com/influxdata/influxdb_iox/pull/6782 now as it won't capture anything

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
I am also going to merge this PR as well because I saw the flaky test hit another PR as well

File: test_helpers_end_to_end/src/snapshot_comparison.rs (hunk 2 above)
Comment (carols10cents) [reply test_helpers_end_to_end/src/snapshot_comparison.rs:105]: Oooooooh I missed that one. It'll still capture the files though?

Comment (alamb) [reply test_helpers_end_to_end/src/snapshot_comparison.rs:105]: > Oooooooh I missed that one. It'll still capture the files though?

I think so . Proposed revert: https://github.com/influxdata/influxdb_iox/pull/6844

**Pull request merged**